### PR TITLE
Fixed Cheap Lighters not Burning You

### LINKED ISSUE #11132: [Bug]: Cheap ligthers don't burn your hands anymore
### BYOND Version

514.1589

### Round Date

2024/06/29

### Round ID

49149

### Test merges

[#9124: 'Enable serverside 515.1633' by Crossedfall at commit 3e63a01a0a](https://www.github.com/beestation/beestation-hornet/pull/9124)
[#10372: 'Updates bleeding' by PowerfulBacon at commit 5a0a91fd6a](https://www.github.com/beestation/beestation-hornet/pull/10372)
[#11107: 'Makes revolution objectives softer' by PowerfulBacon at commit dfc9b6e023](https://www.github.com/beestation/beestation-hornet/pull/11107)
[#11128: 'Return of Stat icons' by PowerfulBacon at commit 9f7f9f66e0](https://www.github.com/beestation/beestation-hornet/pull/11128)

### Describe the bug

Using the cheap lighter without gloves doesn't burn you anymore

### What did you expect?

That you will get burned and a mood debuff.

### Reproducible steps

1. spam use a cheap lighter without any gloves on

### PR DESCRIPTION
**Pull request by MarkusLarsson421**
## About The Pull Request
Fixes #11132.

Simplifies the code and changes a FALSE to TRUE.

## Why It's Good For The Game
Fixes a bug.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

image

image

</details>

## Changelog
:cl:
fix: Fixes cheap lights not burning you
/:cl:

### CODE REVIEW

File: code/game/objects/items/cigs_lighters.dm [hunk 1]
```diff
@@ -709,12 +709,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 				var/prot = FALSE
 				var/mob/living/carbon/human/H = user
 
-				if(istype(H) && H.gloves)
-					var/obj/item/clothing/gloves/G = H.gloves
-					if(G.max_heat_protection_temperature)
-						prot = (G.max_heat_protection_temperature > 360)
-				else
-					prot = TRUE
+				prot = istype(H) && H.gloves ?  TRUE : FALSE
```
Comment (PowerfulBacon) [inline code/game/objects/items/cigs_lighters.dm]:
```suggestion
				prot = istype(H) && H.gloves
```

File: code/game/objects/items/cigs_lighters.dm [hunk 2]
```diff
@@ -709,12 +709,7 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 				var/prot = FALSE
 				var/mob/living/carbon/human/H = user
 
-				if(istype(H) && H.gloves)
-					var/obj/item/clothing/gloves/G = H.gloves
-					if(G.max_heat_protection_temperature)
-						prot = (G.max_heat_protection_temperature > 360)
-				else
-					prot = TRUE
+				prot = istype(H) && H.gloves
```
Comment (PowerfulBacon) [inline code/game/objects/items/cigs_lighters.dm]: Move this to where the variable declaration is.
var/prot = ...

**Review by PowerfulBacon — APPROVED** [review]

File: code/game/objects/items/cigs_lighters.dm [hunk 3]
```diff
@@ -706,15 +706,9 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
-				var/prot = FALSE
 				var/mob/living/carbon/human/H = user
 
-				if(istype(H) && H.gloves)
-					var/obj/item/clothing/gloves/G = H.gloves
-					if(G.max_heat_protection_temperature)
-						prot = (G.max_heat_protection_temperature > 360)
-				else
-					prot = TRUE
+				var/prot = istype(H) && H.gloves
```
Comment (PowerfulBacon) [inline code/game/objects/items/cigs_lighters.dm]:
```suggestion
				var/prot = !istype(H) || H.gloves
```

**Review by PowerfulBacon — DISMISSED** [review]
This is needed for logical correctness, if you are not a human then you will not be protected unlike before. The previous code from before this PR was bugged though and would protect you if you didn't have gloves on.

File: code/game/objects/items/cigs_lighters.dm (hunk 3 above)
Comment (PowerfulBacon) [reply code/game/objects/items/cigs_lighters.dm]: Unless you want it so that monkeys will always get burnt, which is also fine

**Review by Rukofamicom — APPROVED** [review]

**Pull request merged**